test/event_test.dart

```diff
@@ -1200,6 +1212,581 @@ void main() {
+        isEdit: false,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: false,
+      );
+      testUnlocalizedBody(
+        expectation: 'm.room.message',
+        plaintextBody: false,
+        body: 5,
+        formattedBody: '<b>formatted body</b>',
+        html: false,
+        isEdit: false,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: false,
+      );
+      testUnlocalizedBody(
+        expectation: 'body',
+        plaintextBody: false,
+        body: 'body',
+        formattedBody: '<b>formatted body</b>',
+        html: false,
+        isEdit: true,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: false,
+      );
+      testUnlocalizedBody(
+        expectation: 'body',
+        plaintextBody: false,
+        body: 'body',
+        formattedBody: null,
+        html: true,
+        isEdit: false,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: false,
+      );
+      testUnlocalizedBody(
+        expectation: 'body',
+        plaintextBody: true,
+        body: 'body',
+        formattedBody: '<b>formatted body</b>',
+        html: false,
+        isEdit: false,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: false,
+      );
+      testUnlocalizedBody(
+        expectation: 'body',
+        plaintextBody: true,
+        body: 'body',
+        formattedBody: null,
+        html: true,
+        isEdit: false,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: false,
+      );
+      testUnlocalizedBody(
+        expectation: 'body',
+        plaintextBody: true,
+        body: 'body',
+        // do we actually expect this to then use the body?
+        formattedBody: '',
+        html: true,
+        isEdit: false,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: false,
+      );
+      testUnlocalizedBody(
+        expectation: 'body',
+        plaintextBody: true,
+        body: 'body',
+        formattedBody: 5,
+        html: true,
+        isEdit: false,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: false,
+      );
+      testUnlocalizedBody(
+        expectation: 'body',
+        plaintextBody: false,
+        body: 'body',
+        formattedBody: '<b>formatted body</b>',
+        html: true,
+        isEdit: true,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: false,
+      );
+      testUnlocalizedBody(
+        expectation: 'body',
+        plaintextBody: false,
+        body: 'body',
+        formattedBody: '<b>formatted body</b>',
+        html: true,
+        isEdit: true,
+        editBody: null,
+        editFormattedBody: null,
+        editHtml: true,
+      );
+      testUnlocalizedBody(
+        expectation: '**formatted body**',
+        plaintextBody: true,
+        body: 'body',
+        formattedBody: '<b>formatted body</b>',
```

Comment:
same here, would expect editBody

Reply:
Edit body is null, which is an invalid value, so it falls back to using the event formatted body. So you expect an empty string here? Even though the newContent is missing?

Reply:
> So you expect an empty string here?

yep, falling back to the old event sounds bad

Reply:
It's not the old event, it is the fallback body in the new event, it just usually has a weird `*` in front of it or so.

Reply:
ah right that makes sense